### PR TITLE
Improve documentation for connect

### PR DESCRIPTION
**Pull request by mzaffalon**

### CODE REVIEW

**Comment by mzaffalon** [comment]
I find the part related to the `C` block confusing, since it has one input according to the state space representation but two inbound arrows according to the drawing. What about having the output `yP` meet the line `yR` since there is the `addP` block in between, in the same way as for the `uP` block?

**Comment by mzaffalon** [comment]
Now that I see: the function's signature in the documentation appears to be different from the code:
```
connect(systems, connections; w1, z1 = (:), verbose = true, kwargs...)
connect(systems; u1::Vector{Symbol}, y1::Vector{Symbol}, w1, z1 = (:), verbose = true, kwargs...)
```
Are these standard names? Can we have `connections`, `u` and `y` for the inputs and outputs of the new system?

Also in the function documentation: "`z1`: outputs of the constructed system (can overlap with `y1`)": `y1` is not in the function's signature that is called at line https://github.com/JuliaControl/RobustAndOptimalControl.jl/blob/master/src/named_systems2.jl#L466.

**Comment by baggepinnen** [comment]
Thanks for your PR :) 

> What about having the output yP meet the line yR since there is the addP block in between, in the same way as for the uP block?

Yes that sounds very reasonable.

> Also in the function documentation: "z1: outputs of the constructed system (can overlap with y1)": y1 is not in the function's signature that is called at line

The interface where the user passed `u1,y1` predates the method that takes `connections::Vector{Pair}`, and it looks like the docstring hasn't been fully updated to reflect this. We can probably just remove the ` (can overlap with y1)` since there is now no chance that the user interprets `y1` as outputs of the resulting system when its listed as a `y => u` connection pair instead.

**Comment by mzaffalon** [comment]
> The interface where the user passed u1,y1 predates the method
I should have guessed that by the file name `named_systems2.jl`.

File: src/named_systems2.jl [hunk 1]
```diff
@@ -401,15 +428,15 @@ The following complicated feedback interconnection
...
-────►   F   ├─┴──►   R   │      │    C    ├────+────►   P   ├────┬────►
-    │       │    │       │   ┌──►         │         │       │    │
-    └───────┘    └───────┘   │  └─────────┘         └───────┘    │
-                             │                                   │
-                             └───────────────────────────────────┘
+               ┌────────────────────────────────┐
+               │                                │
+    ┌───────┐  │  ┌───────┐        ┌───────┐    │    ┌───────┐
+uF  │       │  │  │       | yR     │       │ yC │ uP │       │   yP
+────►   F   ├──┴──►   R   │────┬───►   C   ├────┴────►   P   ├───┬────►
```
Comment (baggepinnen) [inline src/named_systems2.jl:435]:
```suggestion
────►   F   ├──┴──►   R   │────+───►   C   ├────+────►   P   ├───┬────►
```
it probably makes sense to keep the `+`es?

File: src/named_systems2.jl [hunk 2]
```diff
@@ -380,18 +380,45 @@ end
...
-- `connections`: a vector of pairs indicating output => input mappings. Each pair in this vector maps an output to an input. Each output must appear as an output in one of `systems`, and similarly each input must appear as an input in one of `systems`. All outputs must have unique names, but an input may have the same name as an output. In the example below the connection `:uP => :uP` connects the output `:uP` of the `addP` block to `P`'s input `:uP`
-- `w1`: external signals that appear as inputs in the constructed system, `:uF` in the example below, use `(:)` to indicate all signals
-- `z1`: outputs of the constructed system (can overlap with `y1`), use `(:)` to indicate all signals
+- `connections`: a vector of pairs output => input, where aach pair maps an output to an input. Each output must appear as an output in one of `systems`, and similarly each input must appear as an input in one of `systems`. All inputs must have unique names and so must all outputs, but an input may have the same name as an output. In the example below the connection `:uP => :uP` connects the output `:uP` of the `addP` block to `P`'s input `:uP`
```
Comment (baggepinnen) [inline src/named_systems2.jl:419]:
```suggestion
- `connections`: a vector of pairs output => input, where each pair maps an output to an input. Each output must appear as an output in one of `systems`, and similarly each input must appear as an input in one of `systems`. All inputs must have unique names and so must all outputs, but an input may have the same name as an output. In the example below the connection `:uP => :uP` connects the output `:uP` of the `addP` block to `P`'s input `:uP`
```

**Comment by baggepinnen** [comment]
Thanks for your contribution :)

**Comment by mzaffalon** [comment]
You are kidding? You are doing a hell of a job!

**Pull request merged**